src/NuGet.Core/NuGet.Configuration/PackageSource/PackageSourceProvider.cs

```diff
@@ -698,7 +709,8 @@ public void SavePackageSources(IEnumerable<PackageSource> sources, PackageSource
 
                 if (existingSettingsLookup != null &&
                     existingSettingsLookup.TryGetValue(source.Name, out existingSourceItem) &&
-                    ReadProtocolVersion(existingSourceItem) == source.ProtocolVersion)
+                    ReadProtocolVersion(existingSourceItem) == source.ProtocolVersion &&
+                    ReadAllowInsecureConnections(existingSourceItem) == source.AllowInsecureConnections)
```

Comment:
This is buggy, and exactly what I'm fixing in https://github.com/NuGet/NuGet.Client/pull/5334

Unfortunately, although I addressed all feedback 2 days ago, nobody has reviewed or approved the PR since then, so it's still not merged 😞

Reply:
![image](https://github.com/NuGet/NuGet.Client/assets/2878341/23db2c45-1272-4847-97ed-2022b0c049ed)

We didn't know it was ready for a new review :)

Reply:
@zivkan can you reach out to @aortiz-msft directly?
![image](https://github.com/NuGet/NuGet.Client/assets/49205731/0fe0ea66-6a5d-454c-be48-e7e670321bfe)

Reply:
Thanks for reminding me about this recent change! This is fixed.